plugins/okta/pkg/app/fetch.go

```diff
@@ -14,119 +14,208 @@ import (
 )
 
 type FetchCmd struct {
-	Domain   string `env:"DS_OKTA_DOMAIN"`
-	APIToken string `env:"DS_OKTA_TOKEN"`
-	Groups   bool   `env:"DS_OKTA_GROUPS" default:"true" negatable:""`
-	Roles    bool   `env:"DS_OKTA_ROLES" default:"true" negatable:""`
-}
+	Domain         string `env:"DS_OKTA_DOMAIN"`
+	APIToken       string `env:"DS_OKTA_TOKEN"`
+	Groups         bool   `env:"DS_OKTA_GROUPS" default:"true" negatable:""`
+	Roles          bool   `env:"DS_OKTA_ROLES" default:"true" negatable:""`
+	RequestTimeout int64  `default:"0" optional:""`
 
-type OktaPager func(context.Context, *okta.Response, *[]*okta.User) (*okta.Response, error)
+	oktaClient oktaclient.OktaClient `kong:"-"`
+}
 
-func (cmd *FetchCmd) Run(ctx *kong.Context) error {
-	oktaClient, err := oktaclient.NewOktaClient(context.Background(), cmd.Domain, cmd.APIToken)
+func (fetcher *FetchCmd) Run(kongCtx *kong.Context) error {
+	ctx := context.Background()
+	oktaClient, err := oktaclient.NewOktaClient(ctx, fetcher.Domain, fetcher.APIToken, fetcher.RequestTimeout)
 	if err != nil {
 		return err
 	}
+	fetcher.oktaClient = oktaClient
 	results := make(chan map[string]interface{}, 1)
 	errors := make(chan error, 1)
 	go func() {
-		cmd.Fetch(oktaClient, results, errors)
+		fetcher.Fetch(ctx, results, errors)
 		close(results)
 		close(errors)
 	}()
 	return plugin.NewDSPlugin().WriteFetchOutput(results, errors, false)
 }
 
-func (cmd *FetchCmd) Fetch(oktaClient oktaclient.OktaClient, results chan map[string]interface{}, errors chan error) {
-	users, resp, err := oktaClient.ListUsers(context.Background(), nil)
-	if err != nil {
-		errors <- err
-		common.SetExitCode(1)
-		return
-	}
-	err = handleResponse(resp)
-	if err != nil {
-		errors <- err
-		common.SetExitCode(1)
-		return
-	}
+func (fetcher *FetchCmd) Fetch(ctx context.Context, results chan map[string]interface{}, errors chan error) {
```

Comment:
We cold make it a little easier to read by breaking down this method into multiple ones. I also reworked a little the logic for retrieving pages of users

```go
func (fetcher *FetchCmd) Fetch(ctx context.Context, results chan map[string]interface{}, errors chan error) {
	var response *okta.Response

	for {
		var users []*okta.User
		var err error

		users, response, err = fetcher.oktaClient.ListUsers(ctx, nil)
		if err != nil {
			errors <- err
			common.SetExitCode(1)
			return
		}
		err = handleResponse(response)
		if err != nil {
			errors <- err
			common.SetExitCode(1)
			return
		}

		for _, user := range users {
			userResult, err := fetcher.processUser(ctx, user)
			if err != nil {
				errors <- err
			}
			results <- userResult
		}

		if response != nil && response.HasNextPage() {
			response, err = response.Next(ctx, &users)
		} else {
			break
		}
	}
}

func (fetcher *FetchCmd) processUser(ctx context.Context, user *okta.User) (map[string]interface{}, error) {
	userBytes, err := json.Marshal(user)
	if err != nil {
		common.SetExitCode(1)
		return nil, err
	}
	var userResult map[string]interface{}
	err = json.Unmarshal(userBytes, &userResult)
	if err != nil {
		common.SetExitCode(1)
		return nil, err
	}

	if fetcher.Groups {
		// Write all user groups
		groups, err := fetcher.getGroups(ctx, user.Id)
		if err != nil {
			common.SetExitCode(1)
			return nil, err
		}
		userResult["groups"] = groups
	}

	if fetcher.Roles {
		// Write all user roles
		roles, err := fetcher.getRoles(ctx, user.Id)
		if err != nil {
			common.SetExitCode(1)
			return nil, err
		}
		userResult["roles"] = roles
	}
	return userResult, nil
}
```